src/components/ui/AppButton/index.tsx

```diff
@@ -0,0 +1,19 @@
+import React, {useCallback} from 'react';
+import {Text, TouchableOpacity} from 'react-native';
+import {IButtonProps} from '../type/AppButtun-types';
+
+import getStyles from './styles';
+const AppButton = (props: IButtonProps) => {
+  const styles = useCallback(() => getStyles(props), [props])();
```

Comment:
lets rewrite this. lets extract conditional styles into component and combine it here with static styles. like that:
```
const containerStyles = useMemo(
    () => [
      styles.container,
      { backgroundColor: type === 'primary' ? "2D3E6B":"EAEAEA"}
    ],
    [type],
  ); 
```